### PR TITLE
[IMP] hr_attendance: Add extra hours validation mecanism

### PR DESCRIPTION
**Pull request by jugj-odoo**
task-3999208

### CODE REVIEW

File: addons/hr_attendance/models/hr_attendance.py [hunk 1]
```diff
@@ -117,19 +122,38 @@ def _compute_overtime_hours(self):
...
+    def _compute_validated_overtime_hours(self):
+        for attendance in self:
+            attendance.validated_overtime_hours = attendance.overtime_hours
+
+    @api.depends('employee_id')
+    def _compute_overtime_status(self):
         for attendance in self:
-            attendance.overtime_hours = attendance.worked_hours * ((100 - att_progress_values.get(attendance.id, 100))/100)
+            if not attendance.overtime_status:
+                attendance.overtime_status = "to_approve" if attendance.employee_id.company_id.attendance_overtime_validation == 'by_manager' else "approved"
```
Comment (mepe-odoo) [inline addons/hr_attendance/models/hr_attendance.py:156]: this line seems too long ;)

**Review by mepe-odoo — CHANGES_REQUESTED** [review]
Hello; huge work (my brain needs an other coffe :coffee: after this review)

I add some comments; the most are just nickpicking; but one is about your structure choice for a variable :)

File: addons/hr_attendance/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -36,22 +36,15 @@
...
+                            <div class="mt16">
                                 <label for="overtime_company_threshold" class="o_form_label">
                                     Tolerance Time In Favor Of Company
                                 </label>
                                 <div class="text-muted">
                                     Allow a period of time (around working hours) where extra time will not be counted, in benefit of the company
                                 </div>
-                                <span>Time Period </span><field name="overtime_company_threshold" class="text-center"
-                                    required="hr_attendance_overtime" style="width: 10%; min-width: 4rem;"/><span> Minutes</span>
+                                <span>Time Period </span><field name="overtime_company_threshold" class="text-center" style="width: 10%; min-width: 4rem;"/><span> Minutes</span>
```
[the remark this answers is not in the document]
Comment (jugj-odoo) [reply addons/hr_attendance/views/res_config_settings_views.xml:47]: it was like that before, but i'll change it

File: addons/hr_attendance/models/hr_attendance.py [hunk 3]
```diff
@@ -393,6 +415,8 @@ def _update_overtime(self, employee_attendance_dates=None):
         overtime_to_unlink.sudo().unlink()
         self.env.add_to_compute(self._fields['overtime_hours'],
                                 self.search([('employee_id', 'in', employees_worked_hours_to_compute)]))
+        self.env.add_to_compute(self._fields['validated_overtime_hours'],
+                                self.search([('employee_id', 'in', employees_worked_hours_to_compute)]))
```
Comment (tivisse) [inline addons/hr_attendance/models/hr_attendance.py:419]: Seems we could get rid on 1 search call over here

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**